docs/en/latest/discovery/consul_kv.md

```diff
@@ -73,6 +76,31 @@ The `keepalive` has two optional values:
 - `true`, default and recommend value, use the long pull way to query consul servers
 - `false`, not recommend, it would use the short pull way to query consul servers, then you can set the `fetch_interval` for fetch interval
 
+#### Dump Data
+
+When we need reload `apisix` online, as the `consul_kv` module maybe loads data from CONSUL slower than load routes from ETCD, and would get the log at the moment before load successfully from consul:
+
+```
+ http_access_phase(): failed to set upstream: no valid upstream node
+```
+
+So, we import the `dump` function for `consul_kv` module. When reload, would load the dump file before from consul; when the registered nodes in consul been updated, would dump the upstream nodes into file automatically.
+
+The `dump` has three optional values now:
+
+- `path`, the dump file save path
+    - support relative path, eg: `logs/consul_kv.dump`
+    - support absolute path, eg: `/tmp/consul_kv.bin`
+    - make sure the dump file's parent path exist
+    - make sure the `apisix` has the dump file's read-write access permission,eg: `chown  www:root conf/upstream.d/`
+- `load_on_init`, default value is `true`
+    - if `true`, will try load the data from dump file before load data from consul
+    - if `false`, ignore load data
+    - and anyhow, we don't need to prepare a dump file for apisix at anytime
```

Comment:
This sentence confuses me. Does it look like we need to prepare a dump file if load_on_init is true before starting APISIX?

```
local function read_dump_srvs()
    local data, err = util.read_file(dump_params.path)
    if not data then
        log.warn("read dump file get error: ", err)
        return
    end
```

Reply:
What about this ?

- `load_on_init`, default value is `true`
    - if `true`, just try to load the data from the dump file before loading data from    consul when starting
    - if `false`, ignore loading data from the dump file
    - Whether `true` or `false`, we don't need to prepare a dump file for apisix at anytime

Reply:
If `load_on_init` is true, we need to prepare a dump file, otherwise a warning will be logged.

Reply:
Here is my doc:

- `load_on_init`, default value is `true`
    - if `true`, just try to load the data from the dump file before loading data from  consul when starting, does not care the dump file exists or not
    - if `false`, ignore loading data from the dump file
    - Whether `true` or `false`, we don't need to prepare a dump file for apisix at anytime